experiment/measurer/measure_worker.py

```diff
@@ -231,26 +273,40 @@ def archive_crashes(self, cycle):
         gsutil.cp(archive, gcs_path)
         os.remove(archive)
 
-    def update_measured_files(self):
+    def update_measured_files(self, cycle):
         """Updates the measured-files.txt file for this trial with
         files measured in this snapshot."""
         current_files = set(os.listdir(self.corpus_dir))
-        already_measured = self.get_measured_files()
-        filesystem.write(self.measured_files_path,
-                         '\n'.join(current_files.union(already_measured)))
+        previous_files = self.get_prev_measured_files(cycle)
+        all_files = current_files.union(previous_files)
+
+        measured_files_state = self.get_measured_files_state(cycle)
+        measured_files_state.set_current(list(all_files))
+
+        return all_files
 
-    def get_measured_files(self):
-        """Returns a the set of files that have been measured for this
-        snapshot's trials."""
-        if not os.path.exists(self.measured_files_path):
-            return set()
-        return set(filesystem.read(self.measured_files_path).splitlines())
+    def update_state_for_unchanged_cycle(self, cycle):
+        """Update the  covered-pcs and  measured-files state  files so  that the
+        states for |cycle| are the same as |cycle -1|."""
+        state_files = [
+            self.get_covered_pcs_state(cycle),
+            StateFile('measured-files', self.state_dir, cycle)
+        ]
+        for state_file in state_files:
+            prev_state = state_file.get_previous(first_cycle_default=[])
+            state_file.set_current(prev_state)
```

Comment:
Why we do need to set state for unchanged cycle, why we can't use a bool ?

Reply:
Because then we would have to retrieve the previous file's PCs. And then what if that's also unchanged? We could end up doing 20 network requests instead of one. A bool is more space efficient but I think it is not as good.
I think a bool would work well if it was all one file though as I mentioned it could be here:
>Instead of having a measured-files-0001.json, measured-files-0002.json... file for each cycle I suppose we could have one json file where cycles are keys and the value is the measured-files for that cycle. Does this seem like a better design?

WDYT of that solution vs multiple files as I have now?